### PR TITLE
Promote crd to apiextensions.k8s.io/v1

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

This might break compatibility with k8s < 1.19, but is a necessary step to support k8s 1.22+ 

https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]
LGTM, but just to confirm: did you try this in OpenShift as well?

File: Makefile [hunk 1]
```diff
@@ -393,7 +393,7 @@ install-tools:
 		golang.org/x/lint/golint \
 		golang.org/x/tools/cmd/goimports \
 		github.com/securego/gosec/cmd/gosec@v0.0.0-20191008095658-28c1128b7336 \
-		sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 \
+		sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0 \
```
Comment (pavolloffay) [inline Makefile:396]: Did bumping this version update the CRD yamls?

Comment (pavolloffay) [reply Makefile:396]: or did you run any command to generate CRD yalms?

Comment (rubenvp8510) [reply Makefile:396]: I ran the controller-gen command and it generates the new CRD with the version.

**Comment by rubenvp8510** [comment]
> LGTM, but just to confirm: did you try this in OpenShift as well?

Tested with CRC and all-in-one strategy and it works .

**Comment by jpkrohling** [comment]
I think I restarted the tests a couple of times already. If this fails again, it might be a real failure, @rubenvp8510

**Comment by rubenvp8510** [comment]
> I think I restarted the tests a couple of times already. If this fails again, it might be a real failure, @rubenvp8510

Yes, I was investigating about the failures, it seems like I need to migrate the Ingress API too, and because of that I need to bump some versions of the libraries, specifically `k8s.io/api`. I'll request your review again once I finish.

**Review by pavolloffay — APPROVED** [review]
LGTM,

does this PR break compatibility with older k8s clusters?

File: .github/workflows/e2e-kubernetes.yaml [hunk 2]
```diff
@@ -8,28 +8,28 @@ on:
...
     steps:
     - uses: actions/setup-go@v2.1.3
       with:
-        go-version: '1.14.4'
-    - uses: jpkrohling/setup-minikube@v1.0.2
+        go-version: '1.14.12'
+    - uses: actions/checkout@v2.3.4
+    - name: "setup docker"
+      run: ./.ci/setup-docker.sh
+    - uses: manusa/actions-setup-minikube@v2.4.1
```
Comment (pavolloffay) [inline .github/workflows/e2e-kubernetes.yaml:22]: nice I didn't see this. In the future we could also consider using `kind`.

Comment (jpkrohling) [reply .github/workflows/e2e-kubernetes.yaml:22]: Indeed, the new e2e tests are using kind and kuttl :-)

**Comment by rubenvp8510** [comment]
> LGTM,
> 
> does this PR break compatibility with older k8s clusters?

Yes :( 1.19+ is required.

**Pull request merged**